mslib/msui/msui_mainwindow.py

```diff
@@ -468,6 +471,128 @@ def __init__(self, mscolab_data_dir=None, *args):
             self.actionUpdater.triggered.connect(self.updater.show)
         self.openOperationsGb.hide()
 
+        self.StoreLayout.clicked.connect(self.store_operation_data)
+        self.RestoreLayout.clicked.connect(self.restore_operation_data)
+        self.allowed_widget_names = ["TopViewWindow", "SideViewWindow", "TableViewWindow", "LinearWindow"]
+        self.listOperationsMSC.itemChanged.connect(self.operation_changed)
+
+    def operation_changed(self):
+        folder_path = MSUI_CONFIG_PATH
+        file_name = self.listOperationsMSC.currentItem()
+        file_path = os.path.join(folder_path, file_name.text() + ".txt")
+        if os.path.exists(file_path):
+            result = QtWidgets.QMessageBox.question(self, "Saved Layout Detected", "Would you like to restore it?",
+                                                    QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No, QtWidgets.QMessageBox.Yes)
```

Comment:
after a cycle login/logout I got into a loop of Yes/No answers.

Reply:
If I open up a operation for which I previously saved a layout I get asked to restore it. If I answer yes the windows are restored, but then I am asked again if I want to restore. So I get a loop as well.